auto_update_tests.py

```diff
@@ -76,22 +76,35 @@
 
 print '\n[ checking wasm-link-metadata testcases... ]\n'
 
+link_metadata_extension_arg_map = {
+  '.json': [],
+  '.jscall.json': ['--emscripten-reserved-function-pointers=3'],
+}
 for obj_path in files_with_pattern('test', 'lld', '*.o'):
   print '..', obj_path
-  json_path = obj_path.replace('.o', '.json')
-
-  cmd = WASM_LINK_METADATA + [obj_path]
-  actual = run_command(cmd)
-  with open(json_path, 'w') as o: o.write(actual)
+  for ext, ext_args in link_metadata_extension_arg_map.items():
```

Comment:
Nit: `items` could be `iteritems`. This is totally moot for this case (allocating a list of length 2 is a non-issue), but as far as I know, in general iterating over generator expressions is preferred to iterating over a list (e.g. `range` vs. `xrange`)

See also: https://stackoverflow.com/questions/10458437/what-is-the-difference-between-dict-items-and-dict-iteritems

Reply:
The reason I used `items` is `iteritems` is deprecated in Python 3 and `items` does not generate an array but rather generates a generator in Python 3. (So `items` behaves the same way as in `iteritems` does in Python 2) Migrating all scripts to be Python 3 compatible should be something we should do later though.